Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -328,19 +328,35 @@ bool CStaticFunctionDefinitions::DestroyElement(CElement* pElement)
 
     // We can't destroy the root or a player/remote client/console
     int iType = pElement->GetType();
-    if (pElement == m_pMapManager->GetRootElement() || iType == CElement::PLAYER || iType == CElement::CONSOLE ||
+    if (pElement == m_pMapManager->GetRootElement() || iType == CElement::PLAYER || iType == CElement::CONSOLE || 
         g_pGame->GetResourceManager()->IsAResourceElement(pElement))
     {
         return false;
     }
 
+    if (iType == CElement::TEAM) { // Its team trigger onPlayerTeamChange for each player in the team
+        CTeam* pTeam = static_cast<CTeam*>(pElement);
+
+        auto iterBegin = pTeam->PlayersBegin();
+        auto iterEnd = pTeam->PlayersEnd();
+
+        for (auto iter = iterBegin; iter != iterEnd; ++iter)
+        {
+            CPlayer* player = *iter;
+            CLuaArguments arguments;
```

Comment:
Potentially the count of players in a team can be significant. And you can handle this situation better by taking `CLuaArguments arguments` out of the loop and calling `arguments.DeleteArguments()` on an every iteration.

Reply:
Didn't know it was somewhat expensive (good to know)